pkg/local_object_storage/engine/inhume.go

```diff
@@ -227,6 +234,23 @@ func (e *StorageEngine) inhumeAddr(addr oid.Address, prm shard.InhumePrm) (bool,
 
 	prm.SetTargets(append(children, addr)...)
 
+	if shardWithObject != "" {
+		sh := e.getShard(shardWithObject)
+
+		_, err := sh.Inhume(prm)
+		if err != nil {
+			if !errors.Is(err, logicerr.Error) {
+				e.reportShardError(hashedShard(sh), "could not inhume object in shard", err)
+			}
+
+			return false, err
+		}
+
+		return true, nil
+	}
+
+	// has not found the object on any shard, so mark it as inhumed on the most probable one
```

Comment:
How can it happen?

Reply:
tombstones to the objects that are not here (and never were)? the control service that said to mark this obj as garbage (even if you do not have it yet)?